### ISSUE TITLE
Running on docker

### ISSUE THREAD

**Issue by 3den** [copiousfreetime/launchy#108]
I am using capybara + poltergeist to run functional test but since the container cannot access the host browser we are currently getting this error:

```
Failure/Error: save_and_open_page
     Launchy::CommandNotFoundError:
       Unable to find a browser command. If this is unexpected, Please rerun with environment variable LAUNCHY_DEBUG=true or the '-d' commandline option and file a bug at https://github.com/copiousfreetime/launchy/issues/new
```

 Is it possible to run save_and_open_page from a docker container? How?

**Comment by copiousfreetime** [comment]
Hi @3den,  Launchy attempts to find an installed browser locally on the machine that it is running on. If you are using launchy in a remote machine/heroku/vps/docker scenario it will attempt to launch a web browser process on that container/remote machine where the ruby process is running.

Are you using `Capybara::Session#save_and_open_page` on every test, or is this when a failure happens and you would like the to see the page rendered in its error state?

Would you like to see launchy in the docker container somehow send a message to something on the host (your mac?) to open the browser on your mac on the file that is inside the container?

**Comment by 3den** [comment]
@copiousfreetime I only use save_and_open_page when debugging, I am fine with sending a message to the host (my mac) and the container already shares the project directory with my mac, so on `tmp/capybara` I can already see the rendered pages.

The problem is just because I don't know of how to map the launchy command that happens inside of docker to open the browser in my host OS.

**Comment by copiousfreetime** [comment]
@3den Its not really possible with to have something in a container affect its host environment. That is pretty much the purpose of a container. 

For your specific case, launchy really can't help from within the container. What I might suggest you do: is don't use save and open page, maybe just use `save_page` so that the files get dumped to that shared project directory. Then do one of the following:
* use something like [heel](https://github.com/copiousfreetime/heel) to serve up the tmp directory and view them that way. you'll have to hit reload on the index page or something
* use your favorite directory watcher program (guard/directory_watcher/rego) on your mac to watch the `tmp/capybara` and automatically run launchy from your mac to view the page when it is created.

**Comment by thomasdziedzic** [comment]
Did you end up finding a simple solution for this issue?
Just encountered this when getting the same error and then realizing it was because I had the server running in a container.

**Comment by copiousfreetime** [comment]
closing as a stale issue.

**Comment by kamok** [comment]
Hi, is there any way for Launchy to open the browser on a client's computer? I'm hosting on Heroku, thought I would write a small script with Launchy to help my users open their links in batch but I'm running into the above error. If not, is there an alternative?

**Comment by calebhaye** [comment]
I got this working by following @copiousfreetime's advice.

I installed guard on the system (not the container!)
then added the `guard-shell` gem
````
sudo gem install guard
gem install guard-shell
guard init shell

````

Add the following to the newly generated `Guardfile`

````
guard :shell do
  watch(%r{^capybara-tmp/.+\.html}) do |modified_files|
    `open #{modified_files[0]}`
  end
end
````

In your `docker-compose.yml` under the app you are running against (e.g. `app:`) , map the `tmp/capybara` directory to a local directory named `capybara-tmp`:

````
    volumes:
      - './capybara-tmp:/app/user/tmp/capybara/'
    tty: true
    stdin_open: true
````
_Note that I'm using the heroku/ruby Docker image, so everything is under `/app/user`... this may not be true in your case -- just FYI_

I also added the `capybara-tmp` dir to `.gitignore`

Then, start guard by typing:
`guard`

Then, run your code, hit your debugger, and input `save_page`... when the page is saved, the guard watcher is invoked, and the system's `open` command will open it in it's default app (in my case, Chrome)

Hope it helps.

cc: @kamok
